tools/bq-visualizer/src/app/bq_query_plan.ts

```diff
@@ -195,9 +201,104 @@ export class BqQueryPlan {
     this.ganttData = data;
   }
 
+  /** visualize the progress data */
+  asProgressChart(
+      containerName: string,
+      onSelectHandler:
```

Comment:
I recommend creating an interface for this for better readability. For example:
```
export interface OnSelectHandler<T> {
  (chart: T, data: object): void;
}
```

Also this can be reused in both here and in `asSlotUsageChart` below.